test.cpp

```diff
@@ -1,28 +1,57 @@
+#include "simplecpp.h"
 
 #include <iostream>
 #include <sstream>
 #include <vector>
-#include "simplecpp.h"
 
-int numberOfFailedAssertions = 0;
+static int numberOfSuccessfulAssertions = 0;
+static int numberOfFailedAssertions = 0;
+
+#if defined ( WIN32 )
+#define __func__ __FUNCTION__
+#endif
 
-#define ASSERT_EQUALS(expected, actual)  (assertEquals((expected), (actual), __LINE__))
+#define ASSERT_EQUALS(expected, actual)  (assertEquals((expected), (actual), __func__, __FILE__, __LINE__))
+#define TODO_ASSERT_EQUALS(wanted, current, actual) (todoAssertEquals((wanted), (current), (actual), __func__, __FILE__, __LINE__))
 
-static int assertEquals(const std::string &expected, const std::string &actual, int line)
+static int assertEquals(const std::string &expected, const std::string &actual, const std::string & function, const std::string & , int line)
 {
-    if (expected != actual) {
+    if (expected == actual) {
+        numberOfSuccessfulAssertions++;
+    } else {
         numberOfFailedAssertions++;
         std::cerr << "------ assertion failed ---------" << std::endl;
+        std::cerr << "function " << function << std::endl;
         std::cerr << "line " << line << std::endl;
         std::cerr << "expected:" << expected << std::endl;
         std::cerr << "actual:" << actual << std::endl;
     }
     return (expected == actual);
 }
 
-static int assertEquals(const unsigned int &expected, const unsigned int &actual, int line)
+static int succeeded_todos_counter;
+static int todos_counter;
+static int todoAssertEquals(const std::string &wanted, const std::string &expected, const std::string &actual, const std::string & function, const std::string &file, int line)
```

Comment:
I would like that this is implemented in a separate PR. just add the test cases. however I don't currently see a need for TODO_ASSERT.